GameOfLife.java

```diff
@@ -58,21 +83,68 @@ private static void play(String fileName) {
 	// and the rightmost columns. Thus the actual board is surrounded by a "frame" of zeros. You can think
 	// of this frame as representing the infinite number of dead cells that exist in every direction.
 	// This function assumes that the input file contains valid data, and does no input testing.
-	private static int[][] read(String fileName) {
+	public static int[][] read(String fileName) 
+	{
 		In in = new In(fileName); // Constructs an In object for reading the input file
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
-		//// Replace the following statement with your code.
-		return null;
+		for (int i=1; i<rows+1; i++)
+		{
+			String line = in.readLine();
+			for (int j=1; j<cols+1; j++)
+			{
+				if (line !="")
```

Comment:
Don't need 'if else' here; it makes the code more complex. If 'line' is empty (line == ""), then line.length == 0, and board[i][j] is 0 by default.
Also don't need `if (line.length()>j-1)` check, because we assume the input is correct (it was documented in the comment).